### ISSUE TITLE
[feature request] allow admins to provide reason why an item needs updates before being approved

### ISSUE THREAD

**Issue by iSCJT** [ONEARMY/community-platform#2377]
**Is your feature request related to a problem? Please describe.**
Currently an admin can 'reject' a post but there is no way other than exchanging emails to tell them what needs to be changed or improved before it will be approved.

**Describe the solution you'd like**
Add a field to the how to and research objects that stores moderation comments, if a post is rejected then the admin can add comments, these should be displayed when viewing the how to / research article so that the user can make amendments before resubmitting.  The comments should be hidden once the post is approved.

**Describe alternatives you've considered**
Alternative is for the admin and user to have backwards and forwards emails.

**Additional context**

**Comment by thisislawatts** [comment]
This should be extended to cover Map Pins as well, those are another type of content which require moderator approval.

**Comment by iSCJT** [comment]
@thisislawatts I've added a separate issue for the map pins as I think we should maybe tackle it as part of a more thorough update to how map pins are set, can discuss on Thursday #2668

**Comment by sigolenej** [comment]
Hey guys, do you need anything from us on this for now? 
Is there a design from us needed?
